torch/testing/_internal/jit_utils.py

```diff
@@ -54,6 +55,32 @@ def get_execution_plan(graph_executor_state):
                            'only have one execution plan, got: {}'.format(num_plans))
     return execution_plans[0]
 
+class _AssertRaisesRegexWithHighlightContext(object):
+    """
+    A context that checks an exeption is thrown. The exception class must match
+    given exception class, text must regular expression with proper part of if
+    highlighted.
+    """
+
+    def __init__(self, test_case, exception, regex, highlight):
+        self.test_case = test_case
+        self.exception_type = exception
+        self.regex = regex
+        self.highlight = highlight
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, type, value, traceback):
+        with self.test_case.assertRaisesRegex(self.exception_type, self.regex):
+            if type:
+                raise value
+
+        if self.highlight:
+            FileCheck().check_source_highlighted(self.highlight).run(str(value))
+
+        return True
```

Comment:
Is this return value used anywhere?

Reply:
Yes, it is checked:

https://docs.python.org/3/reference/datamodel.html#object.__exit__